File: 백준/Silver/11399. ATM/ATM.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int n; cin>>n;
    vector<int> v(n);
    for(int i=0; i<n; i++) cin>>v[i];
    
    sort(v.begin(), v.end());
    
    int ans=0;
    int delay=0;
    for(int i=0; i<v.size(); i++) {
        ans+=delay;
        delay+=v[i];
        
        // cout<<ans<<" "<<delay<<'\n';
    }
    
    cout<<ans+delay;
    
    return 0;
}

File: 백준/Silver/11399. ATM/ATM.java
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class Main {
	private Scanner sc = new Scanner(System.in);
	
	private void solve() throws FileNotFoundException {
	    //Scanner sc = new Scanner(new FileInputStream("input.txt"));
	    
	    int n = sc.nextInt();
	    int[] arr = new int[n];
	    
	    for(int i=0; i<n; i++) {
	    	arr[i] = sc.nextInt();
	    }
	    
	    Arrays.sort(arr);
	    
	    int ans = 0;
	    for(int i=0; i<n; i++) {
	    	for(int j=0; j<=i; j++) {
	    		ans+=arr[j];
	    	}
	    }
	    System.out.println(ans);
	    
	    sc.close();
	}

	public static void main(String[] args) throws Exception {
	    new Main().solve();
	}
}